src/rpdk/core/contract/suite/handler_commons.py

```diff
@@ -187,3 +195,57 @@ def test_unordered_list_match(inputs, outputs):
         assert all(input in outputs for input in inputs)
     except KeyError as exception:
         raise AssertionError("lists do not match") from exception
+
+
+def transform_model(
+    input_model, output_model, property_transform_keys, property_transform
+):
+    for prop in property_transform_keys:
+        document_input, _path_input, _parent_input = traverse(
+            input_model, list(prop)[1:]
+        )
+        document_output, _path_output, _parent_output = traverse(
+            output_model, list(prop)[1:]
+        )
+        if document_input != document_output:
+            transformed_property = transform(prop, input_model, property_transform)
+            update_transformed_property(prop, transformed_property, input_model)
+
+
+def transform(property_path, input_model, property_transform):
+    path = set_to_string(property_path)
+    file = open("./transformation.js", "a")
+    file.write("var jsonata =require('jsonata')\n")
+    file.write("var model = {0};\n".format(input_model))
+    property_transform_value = property_transform[path].replace('"', '\\"')
+    file.write('var expression = jsonata("{0}");\n'.format(property_transform_value))
+    file.write("var result = expression.evaluate(model);\n")
+    file.write("console.log(result);\n")
```

Comment:
Let's not write other programming languages inside other programming languages line-by-line. Let's just read  another `.js` file if needed so local tooling like autocompletion, syntax highlighting, static analysis, etc. works

Reply:
use the  [data directory](https://github.com/aws-cloudformation/cloudformation-cli/tree/master/src/rpdk/core/data) if we have anything we want to load in from the package

Reply:
Let's not write other programming languages inside other programming languages line-by-line. Let's just read another .js file if needed so local tooling like autocompletion, syntax highlighting, static analysis, etc. works
> As of now jsonata python wrapper is not present for use. We could write our own python wrapper but, for unblocking this functionality this seems to be the quickest solution

Reply:
use the data directory if we have anything we want to load in from the package
> the ```transformation.js``` file is created in the local directory (which is the provider package). Is there some way to fetch the data directory information? May be I am missing something.

Reply:
specifically asking to not use `file.write()` here. Unrelated to broader discussion of JavaScript usage

Reply:
we can render this as a jinja template if planning on doing this. consider also using tmpfile as well so we aren't leaving files behind after execution.